src/python/dxpy/bindings/dxdataobject_functions.py

```diff
@@ -146,16 +146,32 @@ def get_handler(id_or_link, project=None):
 
 def describe(id_or_link, **kwargs):
     '''
-    :param id_or_link: String containing an object ID or dict containing a DXLink
+    :param id_or_link: String containing an object ID or dict containing a DXLink,
+                       or a list of object ID's or dict's containing a DXLink.
 
     Given an object ID, calls :meth:`~dxpy.bindings.DXDataObject.describe` on the object.
 
     Example::
 
         describe("file-1234")
     '''
-    handler = get_handler(id_or_link)
-    return handler.describe(**kwargs)
+    # If this is a list, extract the ids.
+    if isinstance(id_or_link, list):
```

Comment:
Yeah, maybe sets isn't the best example of a useful sequence type, but it seems reasonable to be somewhat agnostic of the exact thing the user passes in, and return a list that has the same iteration order as whatever collection the user supplied. If the user specified a data structure with an unspecified iteration order, that's their problem.

The dict idea would be a more general solution. If you think that would be useful and you feel like implementing it, go for it. I think what you have here is good though.